Add support for TI TLC59208F

tamisoft opened the pull request:
## Description:
The chip is a 8-BIT FM+ I2C BUS LED DRIVER with
8 open-drain output channels.

Its features include:
- 256 linear levels
- group dimming
- group blinking
- 64 slave addresses
- customizable sub addresses and all call address
- output update on stop or on ACK
- 3.3V or 5V supply with 5V tolerant IO
- no glitch startup
- 50mA / output continuous current up to 17V

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#338

## Checklist:
  - [X] The code change is tested and works locally.
  - [X] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [x] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

tamisoft committed:
Add support for TI TLC59208F

The chip is a 8-BIT FM+ I2C BUS LED DRIVER with
8 open-drain output channels.

Its features include:
- 256 linear levels
- group dimming
- group blinking
- 64 slave addresses
- customizable sub addresses and all call address
- output update on stop or on ACK
- 3.3V or 5V supply with 5V tolerant IO
- no glitch startup
- 50mA / output continuous current up to 17V

OttoWinter commented on esphome/components/tlc59208f/tlc59208f_output.cpp:
```diff
@@ -0,0 +1,157 @@
+#define LDRx_OFF 0x00
+#define LDRx_ON 0x01
+#define LDRx_PWM 0x02
+#define LDRx_GRPPWM 0x03
```
macros are discourages - please convert these to const uint8_t as above.

tamisoft replied:
Resolved in ba4054d

OttoWinter commented on esphome/components/tlc59208f/tlc59208f_output.cpp:
```diff
@@ -0,0 +1,157 @@
+  // This is tricky, because reset has to be sent to a special address
+  Wire.beginTransmission(TLC59208F_SWRST_ADDR >> 1);
+  Wire.write(TLC59208F_SWRST_SEQ, 2);
+  if (Wire.endTransmission() != 0) {
```
Ok I see that the bus has to be accessed directly here. But please use `this->parent_->write_byte` instead, otherwise this component won't work if the user has multiple i2c buses.

tamisoft replied:
resolved in 66375b6

tamisoft committed:
Convert macro to uint8_t

Variables had to be renamed, clang-format would protest against
mixed case in global variable name.

tamisoft committed:
Change gen-call reset to use the correct i2c bus
